shell/platform/darwin/macos/README.md

```diff
@@ -0,0 +1,50 @@
+# Flutter macOS Embedder
+
+This directory contains files specific to the Flutter macOS embedder. The
+embedder is built as a macOS framework that is linked into the target
+application. The framework is composed both of macOS-specific code and code
+shared with the iOS embedder. These can be found in:
+```
+flutter/shell/platform/darwin/common/framework
+flutter/shell/platform/darwin/macos/framework
+```
+
+Additionally, the framework relies on utility code shared across all embedders,
+which is found in:
+```
+flutter/shell/platform/common
+```
+
+To learn more, see the [Engine architecture wiki](https://github.com/flutter/flutter/wiki/The-Engine-architecture).
+
+## Building
+
+Building all artifacts required for macOS occurs in two steps:
+* Host binary build: Builds tooling used by the Flutter tool to build Flutter
+  applications targeting macOS.
+* macOS Target build: Builds the framework that implements the macOS Flutter
+  embedder and exposes public API such as `FlutterViewController` used by
+  Flutter applications.
+
+To build the macOS embedder from the `src/flutter` directory, run the following
```

Comment:
Consider mentioning the setup environment guide as that's a prerequisite. Maybe something like:

```suggestion
Once you've [prepared your environment for engine development](https://github.com/flutter/flutter/wiki/Setting-up-the-Engine-development-environment),
you can build the macOS embedder from the `src/flutter` directory using the following
```